### PR TITLE
Add tests for CommandLineEdit

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

**Comment by hackebrot** [comment]
Fixed the flake8 issue, hope you don't mind.

I personally prefer to not use a class in this particular case, but it's totally up to you!

Looks good to me :+1:

**Comment by The-Compiler** [comment]
Sorry for the long delay, I was a bit busy with [the new release](https://github.com/The-Compiler/qutebrowser/releases/tag/v0.2.0) :smile:

Is there a good reason this is using unittest.mock instead of pytest-mock? I thought we were trying to get rid of unittest.mock? :wink:

**Comment by nicoddemus** [comment]
@hackebrot thanks for fixing flake8, my bad I missed it.

About the class, I usually prefer free functions too, but in this case I figure we will add more tests for other widgets from `miscwidgets`, and classes are a good way to group those tests. :smile:

**Comment by nicoddemus** [comment]
@The-Compiler yeah, not sure why I used `with` at that point... changed to use `pytest-mock`. :smile:

Cheers, :beers:

File: tests/misc/test_miscwidgets.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    """Tests for CommandLineEdit widget."""
+
+    @pytest.yield_fixture
+    def cmd_edit(self, qtbot):
+        """
+        Fixture to initialize a CommandLineEdit and its dependencies,
+        cleaning up afterwards.
+        """
+        command_history = lineparser.LimitLineParser('', '', limit=None)
+        objreg.register('command-history', command_history)
```
Comment (The-Compiler) [inline tests/misc/test_miscwidgets.py]: I don't think `command-history` is needed here - it's only used for `qutebrowser/mainwindow/statusbar/command.py:Command`, but not in `qutebrowser/misc/miscwidgets.py:CommandLineEdit` which is what we're testing here.

**Comment by nicoddemus** [comment]
@The-Compiler you're right, not sure why I put that in the first place...

Cheers :beers:

**Pull request merged**